heron/metricsmgr/src/java/org/apache/heron/metricsmgr/sink/InfluxDBSink.java

```diff
@@ -0,0 +1,281 @@
+ * <a href="https://www.influxdata.com/time-series-platform/influxdb/">InfluxDB</a> time series
+ * database server. Each {@link org.apache.heron.spi.metricsmgr.metrics.MetricsInfo MetricInfo}
+ * instance inside a {@link org.apache.heron.spi.metricsmgr.metrics.MetricsRecord MetricsRecord}
+ * for a given metrics source is converted into a {@link org.influxdb.dto.Point Point} and issued
+ * to the InfluxDB client. Writes to the database can either be as soon as recieved or batched for
+ * higher throughput.
+ */
+public class InfluxDBSink implements IMetricsSink {
+
+  private static final Logger LOG = Logger.getLogger(InfluxDBSink.class.getName());
+
+  /** InluxDB connection instance **/
+  private InfluxDB influxDB;
+  /** Heron Topology ID string **/
+  private String topology;
+  /** Heron Cluster name **/
+  private String cluster;
+  /** The role used to launch the topology **/
+  private String role;
+  /** The environment the topology was launched in  **/
+  private String environ;
+  /** The URL of the InfluxDB server host */
+  private String serverHost;
+  /** The port for the InfluxDB server on the host*/
+  private String serverPort;
+  /** The prefix string appended to the front of all topology database names */
+  private String dbPrefix;
+  /** The name of the database on InfluxDB server that metrics will sent to */
+  private String dbName;
+  /** Indicates if metrics should be written as soon as they are processed or batched according to
+   * the flush frequency.
+    */
+  private boolean batchEnabled;
+
+  // Set the config key names for the metrics_sink.yaml config file
+  static final String SERVER_HOST_KEY = "influx-host";
+  static final String SERVER_PORT_KEY = "influx-port";
+  static final String METRIC_DB_PREFIX_KEY = "influx-db-prefix";
+  static final String DB_USERNAME_KEY = "influx-db-username";
+  static final String DB_PASSWORD_KEY = "influx-db-password";
+  static final String BATCH_PROCESS_KEY = "enable-batch-processing";
+  static final String MAIN_BUFFER_KEY = "main-buffer-size";
+  static final String FAIL_BUFFER_KEY = "fail-buffer-size";
+
+  /**
+   * Initialisation method for the InfluxDB metrics sink.
+   *
+   * @param conf An unmodifiableMap containing basic configuration information parsed from the
+   *             metrics-sinks.yaml config file.
+   * @param context context object containing information on the Topology and Heron system this sink
+   *                is registered with.
+   */
+  public void init(Map<String, Object> conf, SinkContext context) {
+
+    LOG.info("Configuration: " + conf.toString());
+    topology = context.getTopologyName();
+    cluster = context.getCluster();
+    role = context.getRole();
+    environ = context.getEnvironment();
+
+    serverHost = (String) conf.get(SERVER_HOST_KEY);
+
+    // The InfluxDB client expects the host connection string to begin with http or https
+    if(!serverHost.contains("http://") & !serverHost.contains("https://")){
```

Comment:
startsWith() and && might be better?

And please add a space after "if" and before "{"